source/memory/memory.go

```diff
@@ -179,6 +192,71 @@ func detectNv() ([]nfdv1alpha1.InstanceFeature, error) {
 	return info, nil
 }
 
+// detectHugePages checks whether huge pages are enabled on the node
+// and retrieves the configured huge page sizes.
+func detectHugePages() (map[string]string, error) {
+	hugePages := map[string]string{
+		"enabled": "false",
+	}
+
+	basePath := hostpath.SysfsDir.Path("kernel/mm/hugepages")
+	subdirs, err := os.ReadDir(basePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return hugePages, nil
+		}
+		return nil, fmt.Errorf("unable to read huge pages size: %w", err)
+	}
+
+	reg := regexp.MustCompile(`hugepages-(\d+)`)
+	for _, entry := range subdirs {
+		if !entry.IsDir() {
+			continue
+		}
+
+		pageSize, err := getHugePageSize(basePath, entry.Name(), reg)
+		if err != nil {
+			klog.ErrorS(err, "skipping huge page entry", "entry", entry.Name())
+			continue
+		}
+		if pageSize != "" {
+			hugePages["enabled"] = "true"
+			hugePages[pageSize] = "true"
+		}
+	}
+
+	return hugePages, nil
+}
+
+func getHugePageSize(basePath, dirName string, reg *regexp.Regexp) (string, error) {
+	matches := reg.FindStringSubmatch(dirName)
+	if len(matches) != 2 {
+		return "", fmt.Errorf("unexpected directory format: %s", dirName)
+	}
```

Comment:
nit: I'd probably simplify by not using regexp for this simple thing

```suggestion
func getHugePageSize(basePath, dirName string) (string, error) {
	if !strings.HasPrefix(dirname, "hugepages-") {
		return "", fmt.Errorf("unexpected hugepages directory name: %s", dirName)
	}
	sizeFormat := strings.strings.TrimPrefix(dirName, "hugepages-") + "Ki"
```

Optionally we could do this checking/trimming also outside this function and just pass the hugepagesize (as an arg

Reply:
I removed the regexp and change the `getHugePageSize` function to `getHugePageTotalCount`.
The logic is a little bit different than suggested although it's simplified.